### DISCUSSION TITLE
Subdomain definitions in Periodic constraints

### DISCUSSION THREAD

**Discussion by abarun22** [idaholab/moose#25126] [Q&A General]
Dear all,
I am trying one of the constraint methods defined here https://mooseframework.inl.gov/source/constraints/PeriodicSegmentalConstraint.html to assign the periodic boundary conditions for my model whose mesh is contained in an exodus file. I am kind of stuck at how the parameters 'primary_subdomain' and 'secondary_subdomain', can be determined given that my mesh is not generated by moose and that the usage of 'SubdomainIDGenerator' can be in question. Could someone give some insights on how these parameters can be worked out for an external mesh?

FYI, i had to look for various other methods of assigning periodic BC's, as the standard ones does not seem to reproduce periodicity when the mesh is refined. The square block example listed [here](https://github.com/idaholab/moose/discussions/18380) can be looked upon that explains this problem.
Kind regards,
Arun

**Comment by GiudGiud** [comment]
Hello

External meshes should also have subdomains. 
Do all your elements belong in the same subdomain in your mesh?

As mentioned in #18380, refinement is only likely to be supported if the nodes still line up, eg if the refinement is the same on both sides of the periodic boundary

Guillaume

**Reply by abarun22** [reply]
Hi @GiudGiud,
Well, i did create a single block for the whole FE in my model. Is that what you are referring to as subdomain? Looking at this, i guess that you want me to create a block (through curve) each for the periodic boundaries and use that in 'PeriodicSegmentalConstraint'. 
Concerning the refinement, as you can see the nodes still line up on that model in #18380, sadly the results aren't as good as the coarse model.

**Reply by GiudGiud** [reply]
Yes a subdomain is the same thing as a block.
I m not familiar enough with PeriodicSegmentalConstraint

can you describe how you are planning to use it and we can tag an expert to validate that
